### ISSUE TITLE
New 'check_every' value needs pipeline re-creation

### ISSUE THREAD

**Issue by propalparolnapervom** [concourse/concourse#4144]
## Bug Report

Concourse 5.3.0

I can see via `fly -t <target_name> gp -p <pipeline_name>` that new value of `check_every` applied successfully.
But in fact pipeline works with previous `check_every` value.
New value of `check_every` works only after pipeline re-creation.

## Steps to Reproduce

Create initial pipeline with `check_every` parameter:
```
- name: pr
  type: pull-request
  source:
    access_token: <access_token>
    repository: "<repository>"
  check_every: 1m
```
Make sure pipeline works with defined check time.
Update pipeline with new value of `check_every` parameter.
```
- name: pr
  type: pull-request
  source:
    access_token: <access_token>
    repository: "<repository>"
  check_every: 30s
```

## Expected Results

Pipeline checks resource with newly defined `check_every` parameter (`30s`).

## Actual Results

Pipeline checks resource with old `check_every` parameter (`1m`).
Thus, `fly -t <target_name> gp -p <pipeline_name>` shows new `check_every` parameter (`30s`).

Pipeline has to be re-created to start work as expected.

## Version Info

* Concourse version: 5.3.0

**Comment by jama22** [comment]
@propalparolnapervom just to clarify, when you say that the pipeline has to be re-created, do you mean that you did a full on `destroy-pipeline` and then `set-pipeline`?

**Comment by robwhitby** [comment]
I'm seeing the same bug... sometimes at least. Can't see what the difference is between when it works and when it doesn't 😕 

I'm also on 5.3.0.

**Comment by migs** [comment]
I'm seeing this same bug, but on 5.4.0

**Comment by robwhitby** [comment]
ah I think this is not a bug, but a consequence of global resources. 

https://concourse-ci.org/global-resources.html

> Since there will be only one resource check for all resources that have the same configuration, the resource that has the shortest check_every configured will result in its pipeline running the checks for that resource configuration.

**Comment by propalparolnapervom** [comment]
> @propalparolnapervom just to clarify, when you say that the pipeline has to be re-created, do you mean that you did a full on `destroy-pipeline` and then `set-pipeline`?
Yes, `destroy-pipeline`/`set-pipeline`.

**Comment by propalparolnapervom** [comment]
We've upgraded to `5.4.0`. 
Can't reproduce this behavior any more: changing `check_every` parameter works fine without pipeline re-creation ...

**Comment by propalparolnapervom** [comment]
after upgrading to `5.4.0` I tried to reproduce the behavior from reproducing steps and it worked fine. that's how I realized that I removed too much lines from my real case code: 
not only `check_every` is changed in my case, but also `webhook_token` is removed.

sorry for the mess, guys. to keep it at least on this level, I've created [new issue](https://github.com/concourse/concourse/issues/4163). this one is gonna be closed.
